contrib/python/api/skydive/rest/client.py

```diff
@@ -65,11 +71,55 @@ def lookup(self, gremlin, klass=None):
                                                  context=context))
 
         opener = request.build_opener(*handlers)
-        req = request.Request(url,
-                              data.encode(),
-                              {'Content-Type': 'application/json'})
 
+        req = request.Request(url, data=urlencoder.urlencode(data).encode())
         resp = opener.open(req)
+        if resp.getcode() == 200:
+            for cookie in self.cookie_jar:
+                if cookie.name == "authtok":
+                    self.authtok = cookie.value
+
+        return resp
+
+    def request(self, path, data=None):
+        if self.username and not self.authenticated:
+            resp = self.authenticate()
+            if resp.getcode() != 200:
+                self.authenticated = False
+                return resp
+            else:
+                self.authenticated = True
+
+        handlers = []
+        url = "%s://%s%s" % (self.scheme, self.endpoint, path)
+        handlers.append(request.HTTPHandler(debuglevel=self.debug))
+        handlers.append(request.HTTPCookieProcessor(self.cookie_jar))
+
+        if self.scheme == "https":
+            if self.insecure:
+                context = ssl._create_unverified_context()
+            else:
+                context = ssl._create_default_context()
+            handlers.append(request.HTTPSHandler(debuglevel=self.debug,
+                                                 context=context))
+
+        opener = request.build_opener(*handlers)
+
+        headers = {'Content-Type': 'application/json'}
+        req = request.Request(url, data=data, headers=headers)
+
+        resp = opener.open(req)
+        if resp.getcode() == 401:
```

Comment:
In case of 401, we do not get a response code, but rather RuntimeException with 401.
Better to wrap with try..except

Reply:
I'll wrap in order to reset the authentication statut but will let the exception so that the user can handle it